File: DKImagePickerControllerDemo_OC/DKImagePickerControllerDemo_OC/CustomCameraUIDelegate/CustomCamera.m
//
//  CustomCamera.m
//  DKImagePickerControllerDemo_OC
//
//  Created by zm on 2017/7/7.
//  Copyright © 2017年 zm. All rights reserved.
//

#import "CustomCamera.h"

@interface CustomCamera ()


@end

@implementation CustomCamera

- (void)viewDidLoad {
    [super viewDidLoad];
    self.delegate = self;
    self.sourceType = UIImagePickerControllerSourceTypeCamera;
    self.mediaTypes = @[(NSString *)kUTTypeImage, (NSString *)kUTTypeMovie];
    // Do any additional setup after loading the view.
}


#pragma mark -- UIImagePickerControllerDelegate
- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info{
    NSString * mediaType = info[UIImagePickerControllerMediaType];
    if ([mediaType isEqualToString:(NSString *)kUTTypeImage]) {
        UIImage * image = info[UIImagePickerControllerOriginalImage];
        if (self.didFinishCapturingImage) {
            self.didFinishCapturingImage(image);
        }
    }else if ([mediaType isEqualToString:(NSString *)kUTTypeMovie]){
        NSURL * videoURL = info[UIImagePickerControllerMediaURL];
        if (self.didFinishCapturingVideo) {
            self.didFinishCapturingVideo(videoURL);
        }
    }
    
}

- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker{
    if (self.didCancel) {
        self.didCancel();
    }
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: DKImagePickerControllerDemo/CustomCamera.swift
//
//  CustomCamera.swift
//  DKImagePickerControllerDemo
//
//  Created by ZhangAo on 16/3/17.
//  Copyright © 2016年 ZhangAo. All rights reserved.
//

import UIKit
import MobileCoreServices

open class CustomUIDelegate: DKImagePickerControllerDefaultUIDelegate, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
	
	var didCancel: (() -> Void)?
	var didFinishCapturingImage: ((_ image: UIImage) -> Void)?
	var didFinishCapturingVideo: ((_ videoURL: URL) -> Void)?
	
	open override func imagePickerControllerCreateCamera(_ imagePickerController: DKImagePickerController,
	                                                       didCancel: @escaping (() -> Void),
	                                                       didFinishCapturingImage: @escaping ((_ image: UIImage) -> Void),
	                                                       didFinishCapturingVideo: @escaping ((_ videoURL: URL) -> Void)
	                                                       ) -> UIViewController {
		self.didCancel = didCancel
		self.didFinishCapturingImage = didFinishCapturingImage
		self.didFinishCapturingVideo = didFinishCapturingVideo
		
		let picker = UIImagePickerController()
		picker.delegate = self
		picker.sourceType = .camera
		picker.mediaTypes = [kUTTypeImage as String, kUTTypeMovie as String]
		
		return picker
	}
	
	// MARK: - UIImagePickerControllerDelegate methods
	
	open func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
		let mediaType = info[UIImagePickerControllerMediaType] as! String
		
		if mediaType == kUTTypeImage as String {
			let image = info[UIImagePickerControllerOriginalImage] as! UIImage
			self.didFinishCapturingImage?(image)
		} else if mediaType == kUTTypeMovie as String {
			let videoURL = info[UIImagePickerControllerMediaURL] as! URL
			self.didFinishCapturingVideo?(videoURL)
		}
	}
	
	open func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
		self.didCancel?()
	}
	
}
